scan/lib/scan/test_command_generator.rb

```diff
@@ -31,8 +31,9 @@ def options
 
         options = []
         options += project_path_array
-        options << "-sdk '#{config[:sdk]}'" if config[:sdk]
-        options << destination # generated in `detect_values`
+        options << "-sdk '#{config[:sdk]}'" if config[:sdk] && !config[:build_for_testing]
+        options << destination unless config[:build_for_testing] # generated in `detect_values`
+        options << "  -sdk iphonesimulator ONLY_ACTIVE_ARCH=NO" if config[:build_for_testing]
```

Comment:
@nikolaykasyanov Different sims wouldn't require different ARCH would they? it's all x86_64 right?

This option will slow down compile times, rather than speed them up. Which sort of defeats the point.

If you want multiple ARCHS, you would specify them in the `device` argument to scan.